### ISSUE TITLE
 webscoketserver start  failed(Address already in use) when after restart instance 

### ISSUE THREAD

**Issue by icecooly** [TooTallNate/Java-WebSocket#981]
**Describe the bug**
[2020-02-22 17:51:57,449] [ERROR] [WebEventServer] [WebSocketSelector-49]- Address already in use
 java.net.BindException: Address already in use
        at java.base/sun.nio.ch.Net.bind0(Native Method) ~[?:?]
        at java.base/sun.nio.ch.Net.bind(Net.java:433) ~[?:?]
        at java.base/sun.nio.ch.Net.bind(Net.java:425) ~[?:?]
        at java.base/sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:225) ~[?:?]
        at java.base/sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:74) ~[?:?]
        at java.base/sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:67) ~[?:?]
        at org.java_websocket.server.WebSocketServer.doSetupSelectorAndServerThread(WebSocketServer.java:505) [CornerstoneBizSystem.jaz:build202002221725]
        at org.java_websocket.server.WebSocketServer.run(WebSocketServer.java:315) 

setReuseAddr(true);
when stop instance will call
webEventServer.stop();

**Comment by PhilipRoman** [comment]
To investigate port issues, use the command `netstat -ab` on Windows, or `netstat -ap` on Linux. This will show you the current status of each port and what process is using it.

If the port is still used by your program and has the status TIME_WAIT, make sure you're correctly setting the 'reuse' flag - **it must be set before starting the server**.

If all of this fails, you need to **show us an example application which reproduces this issue**.

**Comment by icecooly** [comment]
the port is use by webEventServer

but after stop webEventServer   port will existed several seconds

**Comment by nospam2k** [comment]
I'm sorry to reopen this but I didn't want to start another issue. I'm having this problem but can you please provide an example of a correct way of how to wait/retry when the port is still in process of timing out? I've used this library for many years and have only recently run into this "address in use" error when I updated the library. Thank you.

**Comment by PhilipRoman** [comment]
You're right - `setReuseAddr` is all that is needed to fix this issue.
